cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DatabaseBrowser/index.js

```diff
@@ -97,48 +113,47 @@ export default class DatabaseBrowser extends Component {
         }
       );
   }
+
   fetchTables() {
-    let {userName, password, connectionString} = this.state.properties;
-    if (isEmpty(userName) || isEmpty(password) || isEmpty(connectionString)) {
-      return;
-    }
-    let {selectedNamespace: namespace} = NamespaceStore.getState();
+    if (!this.state.connectionId) { return null; }
+
+    let namespace = NamespaceStore.getState().selectedNamespace;
     let params = {
-      namespace
+      namespace,
+      connectionId: this.state.connectionId
     };
-    DataPrepApi
-      .listTables(params, {userName, password, connectionString})
+
+    DataPrepApi.listTables(params)
       .subscribe(
         (res) => {
           this.setState({
-            tables: res,
+            tables: res.values,
             loading: false
           });
         },
         (err) => {
           this.setState({
-            error: typeof err === 'object' ? err.response : err,
+            error: typeof err === 'object' ? err.response.message : err,
```

Comment:
`objectQuery` would be safer here as we are referencing two levels deep in the object. Could have a default error message in case we didn't receive one from backend.